cs/test/core/Util.cs

```diff
@@ -271,6 +273,22 @@ public static ArraySegment<byte> SerializeUnsafeCB<T>(T obj)
             return output.Data;
         }
 
+        public static ArraySegment<byte> SerializeUnsafeCBWithPtr<T>(T obj)
```

Comment:
This is a good point. I don't think we need all combination but we should add another rountrip delegate which serializes to `ArraySegment<byte>` and deserializes using ` byte*`. This shouldn't add much complexity.